### ISSUE TITLE
Restic Volume backups are not working as expected on k8s cluster backed by ceph rbd

### ISSUE THREAD

**Issue by nallurichandu99** [vmware-tanzu/velero#4361]
**What steps did you take and what happened:**

I am trying to use velero and restic for the cluster backup excluding a few namespaces. The cluster was built using kubespray backed by ceph rbd  for persistent storage.

I have installed the velero using helm chart version 2.23.7 and I am using minio as s3 storage interface for the backups and restores. The backup is partially successful but when I verified the backup size its not even close to the actual size.

I have used the following command to start the backup

`velero backup create cluster-backup-11-26 --snapshot-volumes=true --storage-location aws --exclude-namespaces=logging,kube-system,ingress-nginx --include-cluster-resources=true`

For example I have a total of 7G used space on the provisioned volumes but I see only 4M of total size in the s3 bucket.
I see list of all pvs in the backup details but the size is not getting reflected and the volumes are not getting backed up

Also if I backup a single namespace with 2G of PV storage it is able to backup and restore as expected.

**What did you expect to happen:**
All the persistent volumes needs to be backed up via restic

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com/nallurichandu99/57261fa533cdf282405c8288c8c40bdc) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

Note: I see the backup is initiated saying it will be backed up by restic but not finding it in the logs

time="2021-11-16T22:06:58Z" level=info msg="Backing up item" backup=velero/cluster-backup-11-26 logSource="pkg/backup/item_backupper.go:121" name=data namespace= resource=persistentvolumes
time="2021-11-16T22:06:58Z" level=info msg="Executing takePVSnapshot" backup=velero/cluster-backup-11-26 logSource="pkg/backup/item_backupper.go:405" name=finalist namespace= resource=persistentvolumes
time="2021-11-16T22:06:58Z" level=info msg="Skipping snapshot of persistent volume because volume is being backed up with restic." backup=velero/cluster-backup-11-26 logSource="pkg/backup/item_backupper.go:423" name=data namespace= persistentVolume=finalist-data resource=persistentvolumes

**Environment:**

- Velero version (use `velero version`):  
**v1.7.0**

- Velero features (use `velero client config get features`): 
**features: <NOT SET>**

- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T18:52:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:43:56Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

- Kubernetes installer & version:
**kubespray & v2.14.2**

- Cloud provider or hardware configuration:

- OS (e.g. from `/etc/os-release`):

NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by reasonerjt** [comment]
@nallurichandu99 Thanks for the issue! 
Since you are using v1.7, could you use `velero debug` to generate a log bundle, specifying the bakcup name so the details of your backup will be included.

**Comment by nallurichandu99** [comment]
> @nallurichandu99 Thanks for the issue! Since you are using v1.7, could you use `velero debug` to generate a log bundle, specifying the bakcup name so the details of your backup will be included.
Hello, @reasonerjt thanks for the quick response. I got the dump ready. Is there any private way that I can share the dump with you

**Comment by blackpiglet** [comment]
@nallurichandu99 
Do you have Slack account? Maybe we can share contact information or log file on Slack.

**Comment by nallurichandu99** [comment]
> @nallurichandu99 Do you have Slack account? Maybe we can share contact information or log file on Slack. You can find us on #velero-users or #velero-dev channel in Slack.

Hello @blackpiglet, sorry for the delay in response. Yes I do have the Slack account> will reach out to you on the slack channel

**Comment by blackpiglet** [comment]
@nallurichandu99 
Sorry for late response. I checked the log, but the log didn't contain the information your mention `"Skipping snapshot of persistent volume because volume is being backed up with restic."`, and there is no error level logs in velero pod log.
But I found this. It could the reason why PV's data is not backed up.
`time="2021-11-16T22:07:03Z" level=error msg="Error uniquely identifying volume path" backup=velero/cluster-backup-11-26 controller=pod-volume-backup error="expected one matching path, got 0" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/controller/pod_volume_backup_controller.go:442" error.function=github.com/vmware-tanzu/velero/pkg/controller.singlePathMatch logSource="pkg/controller/pod_volume_backup_controller.go:222" name=cluster-backup-11-26-2bvfh namespace=velero`

This means Restic DaemonSet couldn't find the volume path on host path. Usually this is caused by the volume is not mounted by the pod.
